x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/CachePopulationSource.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.blobcache;
+
+/**
+ * The places we populate the cache from
+ */
+public enum CachePopulationSource {
+    /**
+     * When loading data from the blob-store
+     */
+    BlobStore,
+    /**
+     * When fetching data from a peer node
+     */
+    Peer
+}
```

Comment:
This got extracted out to make it easier to use in `IndexWithSource` in stateless, you could argue that `CachePopulationReason` should be extracted also for consistency, and I wouldn't argue, but it's not used elsewhere yet so I left it in `BlobCacheMetrics`.